packages/jbrowse-plugin-apollo/src/ApolloInternetAccount/model.ts

```diff
@@ -389,3 +404,127 @@ export type ApolloInternetAccountStateModel = ReturnType<
 >
 export type ApolloInternetAccountModel =
   Instance<ApolloInternetAccountStateModel>
+
+/**
+ * Get server timestamp and save it into session storage
+ * @param apolloInternetAccount - apollo internet account
+ * @returns
+ */
+async function getAndSetServerTime(session: ApolloSession) {
+  const { internetAccounts } = getRoot(session) as AppRootModel
+  const internetAccount = internetAccounts[0] as ApolloInternetAccountModel
+  const { baseURL } = internetAccount
+  const uri = new URL('changes/getTimestamp', baseURL).toString()
+  const apolloFetch = internetAccount.getFetcher({
+    locationType: 'UriLocation',
+    uri,
+  })
+
+  if (apolloFetch) {
+    const response = await apolloFetch(uri, {
+      method: 'GET',
+    })
+    if (!response.ok) {
+      throw new Error(
+        `Error when fetching server timestamp — ${response.status}`,
+      )
+    } else {
+      sessionStorage.setItem('LastSocketTimestamp', await response.text())
+    }
+  }
+}
+
+function openSocket(session: ApolloSession) {
+  if (!socket.hasListeners('COMMON')) {
+    const { internetAccounts } = getRoot(session) as AppRootModel
+    const internetAccount = internetAccounts[0] as ApolloInternetAccountModel
+    const { baseURL } = internetAccount
+    console.log(`User starts to listen "COMMON" at ${baseURL}`)
+    const { notify } = session
+    const token = internetAccount.retrieveToken()
+    if (!token) {
+      throw new Error(`No Token found`)
+    }
+    const { changeManager } = (session as ApolloSessionModel).apolloDataStore
+    socket.on('COMMON', (message) => {
+      // Save the last server timestamp
+      sessionStorage.setItem('LastSocketTimestamp', message.timestamp)
+      if (message.channel === 'COMMON' && message.userToken !== token) {
+        changeManager?.submitToClientOnly(message.changeInfo)
```

Comment:
This should use the existing `submit`. You'll have to turn the changeInfo into a change first, which you can do with:

```ts
import { Change } from 'apollo-shared'

// ... 

const change = Change.fromJSON(message.changeInfo)
changeManager?.submit(message.changeInfo, { submitToBackend: false })

```